### PR TITLE
Cura 11227 zseam support

### PR DESCRIPTION
**Pull request by saumyaj3**
Add support z-seam type to inset order optimization

The implementation of the order optimizer has been updated to account for the z-seam type for support. This includes adjustment of the start point calculation and  Both 'InsetOrderOptimizer' and 'PathOrderOptimizer' classes have been modified to accommodate these changes.

CURA-11227

### CODE REVIEW

File: src/settings/Settings.cpp [hunk 1]
```diff
@@ -513,6 +513,8 @@ EZSeamType Settings::get<EZSeamType>(const std::string& key) const
         return EZSeamType::SHARPEST_CORNER;
     case "plugin"_sw:
         return EZSeamType::PLUGIN;
+    case "support"_sw:
+        return EZSeamType::SUPPORT;
```
Comment (casperlamboo) [inline src/settings/Settings.cpp:517]: `SUPPORT` is not seam strategy. I wouldn't add it like this. Support should add a condition where the seam is not allowed to be placed. This should be combined with a steam stragegy but is not a stream strategy in it's own right.

Having disallowed areas for seam placement will be very handy in the future aswell for features like "no support in overhangs" etc.

File: include/PathOrderOptimizer.h [hunk 2]
```diff
@@ -604,6 +678,52 @@ class PathOrderOptimizer
...
+
+    size_t pathIfzeamSupportIsCloseToModel(size_t best_pos, const OrderablePath& path, size_t number_of_paths_analysed)
+    {
+        size_t path_size = path.converted_->size();
+        if (path_size > number_of_paths_analysed)
+        {
+            if (! mesh_paths_.empty())
+            {
+                Point2LL current_candidate = (path.converted_)->at(best_pos);
+                if (isVertexCloseToPolygonPath(current_candidate))
```
Comment (casperlamboo) [inline include/PathOrderOptimizer.h:709]: I think this check can be done simpler without introducing the `isVertexCloseToPolygonPath`. 
Take the model-polygon, extend this polygon (with a clipper offset operation) with the `support_z_seam_min_distance` value. Then simply perform a `isInside` check with the candidate point.

**Review by casperlamboo — CHANGES_REQUESTED** [review]

File: src/settings/Settings.cpp (hunk 1 above)
Comment (saumyaj3) [reply src/settings/Settings.cpp]: removed SUPPORT type z sea here: https://github.com/Ultimaker/CuraEngine/pull/2067/commits/70e33aa13c2d81b109b6bee8a4d898f4d1659724

File: include/PathOrderOptimizer.h (hunk 2 above)
Comment (saumyaj3) [reply include/PathOrderOptimizer.h]: I added a function https://github.com/Ultimaker/CuraEngine/pull/2067/commits/70e33aa13c2d81b109b6bee8a4d898f4d1659724 . It creates really nice offset(blue) for the polygon(green) for disallowed area.. Thanks for suggestion :D

**Review by casperlamboo — APPROVED** [review]

**Pull request merged**